packages/next/src/build/swc/options.ts

```diff
@@ -513,7 +513,10 @@ export function getLoaderSWCOptions({
     options.cjsRequireOptimizer = undefined
     // Disable optimizer for node_modules in app browser layer, to avoid unnecessary replacement.
     // e.g. typeof window could result differently in js worker or browser.
-    if (options.jsc.transform.optimizer.globals?.typeofs) {
+    if (
+      options.jsc.transform.optimizer.globals?.typeofs &&
```

Comment:
Is there a human-readable const we could introduce here to make this condition easier to reason about?
Ex: `if (shouldDisableOptimizer)`